docs/DeveloperGuide.md

```diff
@@ -1,34 +1,178 @@
+program.
+
+The purpose of this developer guide is to allow any interested contributors, who wish to develop this application
+further or just curious about the workings of this application. This would allow potential contributors to be able to
+dive right in to improving the code, performance, features or even adding new features much more easily due to the
+understanding of the structure of the codebase and implementation of existing features.
+
+### 1.2 Prerequisites
+
+- Java 11 (can be downloaded from [here](https://www.oracle.com/sg/java/technologies/javase-jdk11-downloads.html))
+- IntelliJ Idea (can be downloaded from [here](https://www.jetbrains.com/idea/download/#section=mac))
+
+### 1.3 Setting up
+
+1. Fork the easyLog repository [here](https://github.com/AY2021S2-CS2113T-T09-4/tp) and git clone it to any location on
+   your computer.
+
+2. Open any IDE (IntelliJ Idea preferred as mentioned in the prerequisites) and click `Configure` -> `Project Defaults`
+   -> `Project Structure` -> `New`
+
+3. Next, go to `Import Project` and select the build.gradle file.
+
+4. After opening the project, go to `src` -> `main` -> `java` -> `seedu.easylog` -> EasyLog and right click on the
+   EasyLog class.
+
+5. Upon successful run, the following opening message will be shown:
+
+```
+Hello! I'm easyLog!
+What can I do for you? Enter help to view commands.
+Looking for save data.
+Save data not found.
+```
+
+## 2. Design
+
+### 2.1 Architecture
+
+![Architecture Diagram](https://user-images.githubusercontent.com/57165946/112990632-ddf7e780-9198-11eb-99ec-f29aafbfc04f.png)
+
+
+The Architecture Diagram shown above illustrates the high-level design of easyLog. We will now proceed to explain each
+component and their respective functionalities below.
+
+### 2.2 EasyLog Component
+
+The EasyLog component initializes the app and looks for any available save file to load. It then proceeds to
+continuously take in input from the user until the user types in a exit command.
+
+### 2.3 Storage Component
+
+![Storage Diagram](https://user-images.githubusercontent.com/57165946/112991416-ab022380-9199-11eb-9ead-cdfcd19cf8e5.png)
+
+The storage component deals with
+- loading, saving and creating save data.
+- generation of receipts when order is done.
+
+### 2.4 Ui Component
+
+The ui component deals with interactions with the user by displaying the appropriate messages according to the users
+input.
+
+### 2.5 Parser Component
+
+![Parser Diagram](https://user-images.githubusercontent.com/57165946/112991918-42677680-919a-11eb-8868-79391cfc2c9e.png)
+
+The parser components deals with the input of the user. It makes sense of the users input and executes commands
+according to the input of the user
+
+### 2.6 Command Component
+
+![Command Diagram](https://user-images.githubusercontent.com/57165946/112992782-15679380-919b-11eb-8c46-436ba701a754.png)
```

Comment:
Arrows should be pointing towards the parent class instead of the child. Diagram is too cluttered with confusing arrows and lines. Lines that intersect can be confusing as to where are the ends of the arrow.
![image](https://user-images.githubusercontent.com/71820915/113078451-b4c17080-9205-11eb-9ea3-f153f88e5f81.png)